src/ScrumBoardItBundle/Services/ApiGitHub.php

```diff
@@ -0,0 +1,258 @@
+                array_push($issues, $issue);
+            }
+        }
+        
+        return $issues;
+    }
+
+    /**
+     *
+     * {@inheritdoc}
+     *
+     */
+    public function getSearchFilters(Request $request)
+    {
+        $session = $request->getSession();
+        if ($session->has('filters'))
+            $this->initFilters($session);
+        $searchFilters = $request->get('github_search') ?: array();
+        
+        if (empty($searchFilters['project']))
+            $searchFilters['project'] = null;
+        
+        $searchFilters['projects'] = $this->getProjects();
+        $searchFilters['sprints'] = $this->getSprints($searchFilters['project']);
+        
+        // Initialise sprint even no sprint is selected
+        if (empty($searchFilters['sprint']))
+            $searchFilters['sprint'] = null;
+        
+        $session->set('filters', array(
+            'project' => $searchFilters['project'],
+            'sprint' => $searchFilters['sprint']
+        ));
+        
+        return $searchFilters;
+    }
+
+    /**
+     *
+     * {@inheritdoc}
+     *
+     */
+    public function addFlag(Request $request, $selected)
+    {
+        if (! empty($selected)) {
+            foreach ($selected as $issue) {
+                $url = $this->getBaseApi($request->getSession()
+                    ->get('filters')['project']) . '/issues/' . $issue . '/labels';
+                $content = '["Printed"]';
+                
+                $this->send($url, $content, 1);
+            }
+        }
+    }
+
+    /**
+     *
+     * {@inheritDoc}
+     *
+     */
+    public function getFormType()
+    {
+        return GithubSearchType::class;
+    }
+
+    /**
+     * Return url for searching projects
+     *
+     * @return string
+     */
+    private function getProjectApi()
+    {
+        $api = 'user/repos';
+        
+        return $this->config['host'] . $api;
+    }
+
+    /**
+     * Return url for searching sprints
+     *
+     * @param string $project            
+     * @return string
+     */
+    private function getSprintApi($project)
+    {
+        $api = $this->getBaseApi($project);
+        
+        return $api . '/milestones';
+    }
+
+    /**
+     * Return base url for the api
+     *
+     * @param string $project            
+     * @return string
+     */
+    private function getBaseApi($project)
+    {
+        $api = 'repos/' . $project;
+        
+        return $this->config['host'] . $api;
+    }
+
+    /**
+     * Return url for searching issues
+     *
+     * @param array $searchFilters            
+     * @return string
+     */
+    private function getIssueApi($searchFilters)
+    {
+        if (! empty($searchFilters['project'])) {
+            $api = $this->getBaseApi($searchFilters['project']) . '/issues';
+            $api .= empty($searchFilters['sprint']) ? '' : ('?milestone=' . $searchFilters['sprint']);
```

Comment:
A normal statement is easier to read than a ternary condition.
Especially in this case, with an empty string in the true condition.